### PR TITLE
NIFI-5436 add ability to set the processor name in the ProcessContext when usin…

### PR DESCRIPTION
**Pull request by ottobackwards**
…g the TestRunner

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [-] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [-] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [-] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [-] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-docs/src/main/asciidoc/developer-guide.adoc [hunk 1]
```diff
@@ -2144,9 +2144,10 @@ you can use the Maven dependency:
 </dependency>
 ----
 
-We create a new `TestRunner` by calling the static `newTestRunner` method of the `TestRunners` class
-(located in the `org.apache.nifi.util` package). This method takes a single argument. That argument can
-either be the class of the Processor to test or can be an instance of a Processor.
+We create a new `TestRunner` by calling on of the static `newTestRunner` methods of the `TestRunners` class
```
Comment (alopresto) [inline nifi-docs/src/main/asciidoc/developer-guide.adoc]: "calling on of" -> "calling one of"

Comment (ottobackwards) [reply nifi-docs/src/main/asciidoc/developer-guide.adoc]: oops.  sorry.  fixed.

**Comment by alopresto** [comment]
Reviewing...

File: nifi-docs/src/main/asciidoc/developer-guide.adoc [hunk 2]
```diff
@@ -2144,9 +2144,10 @@ you can use the Maven dependency:
 </dependency>
 ----
 
-We create a new `TestRunner` by calling the static `newTestRunner` method of the `TestRunners` class
-(located in the `org.apache.nifi.util` package). This method takes a single argument. That argument can
-either be the class of the Processor to test or can be an instance of a Processor.
+We create a new `TestRunner` by calling one of the static `newTestRunner` methods of the `TestRunners` class
+(located in the `org.apache.nifi.util` package). These methods take an argument can
```
Comment (alopresto) [inline nifi-docs/src/main/asciidoc/developer-guide.adoc]: Sorry, missed this construct:

"These methods take an argument (can either be the class of the Processor to test or can be an instance of a Processor), and allow the setting of the processor name as well."

(add parentheses or commas to separate the relative clause).

**Comment by alopresto** [comment]
Ran `contrib-check` and all tests pass. If you make the additional change in the documentation, I am a +1.

File: nifi-docs/src/main/asciidoc/developer-guide.adoc (hunk 2 above)
Comment (ottobackwards) [reply nifi-docs/src/main/asciidoc/developer-guide.adoc]: change going in now, please check @alopresto

**Comment by ottobackwards** [comment]
Thanks for the review @alopresto

**Pull request closed without merging**